tests/daemon/test-machine-id-provider.c

```diff
@@ -206,14 +206,16 @@ main (gint                argc,
 
   ADD_CACHE_TEST_FUNC ("/machine-id-provider/new-succeeds",
                        test_machine_id_provider_new_succeeds);
-  ADD_CACHE_TEST_FUNC ("/machine-id-provider/can-get-id",
-                       test_machine_id_provider_can_get_id);
-  ADD_CACHE_TEST_FUNC ("/machine-id-provider/can-get-id-override",
-                       test_machine_id_provider_can_get_id_override);
-  ADD_CACHE_TEST_FUNC ("/machine-id-provider/can-write-correctly-formed-override-id",
-                       test_machine_id_provider_writes_correctly_formed_override_id);
-  ADD_CACHE_TEST_FUNC ("/machine-id-provider/can-get-id-override-malformed",
-                       test_machine_id_provider_can_get_id_override_malformed);
+  ADD_CACHE_TEST_FUNC ("/machine-id-provider/can-get-tracking-id",
+                       test_machine_id_provider_can_get_tracking_id);
+  ADD_CACHE_TEST_FUNC ("/machine-id-provider/can-write-correctly-formed-tracking-id",
+                       test_machine_id_provider_writes_correctly_formed_tracking_id);
+  ADD_CACHE_TEST_FUNC ("/machine-id-provider/read-malformed-tracking-id",
+                       test_machine_id_provider_read_malformed_tracking_id);
+  /* Test when we do not write a tracking ID file explicitly while setup().
+   * emer_machine_id_provider_new_full should initialize a new tracking ID on its own. */
+  g_test_add ("/machine-id-provider/create-tracking-id-if-unavailable", MachineIdTestFixture, NULL,
```

Comment:
I'd prefer for this to use `setup` just like the others – half of the test is boilerplate copied from `setup`. You could pass GINT_TO_POINTER (TRUE) as the `tdata` parameter (`NULL` here) to tell `setup` not to write the file.

Reply:
Addressed.